lib/ValidateDirectiveVisitor.ts

```diff
@@ -219,8 +219,25 @@ const checkMustValidateInput = (
   if (finalType instanceof GraphQLInputObjectType) {
     const fieldsToCheck = Object.values(finalType.getFields()).filter(field => {
       if (field.type instanceof GraphQLList) {
+        if (field.type.ofType instanceof GraphQLNonNull) {
+          return finalType !== field.type.ofType.ofType;
+        }
+
         return finalType !== field.type.ofType;
       }
+
+      if (field.type instanceof GraphQLNonNull) {
+        if (field.type.ofType instanceof GraphQLList) {
+          if (field.type.ofType.ofType instanceof GraphQLNonNull) {
+            return finalType !== field.type.ofType.ofType.ofType;
+          }
+
+          return finalType !== field.type.ofType.ofType;
```

Comment:
@bfoder this line is missing coverage, see the CI test or run it locally (you can open the html report).

BTW, it would be nice to squash all the commits in a single one, and you could keep the previous test and add the variations below:

```
              a: [TestInput] # existing
              b: [TestInput]!
              c: [TestInput!]
              d: [TestInput!]!
```

Reply:
note that the changed test does not reach this line, it's just to check recursion... you need to test the actual field validation to reach it.

Reply:
Ok, will do ;)